src/auth/ntlm/SMB_LM/ntlm_smb_lm_auth.cc

```diff
@@ -317,33 +316,58 @@ ntlm_check_auth(ntlm_authenticate * auth, int auth_length)
                   user,ntencoded_empty_pass,tmp.str,tmp.l);
             if (memcmp(tmp.str,lmencoded_empty_pass,ENCODED_PASS_LEN)==0) {
                 fprintf(stderr,"ERROR: Empty NT password supplied for user %s\\%s. No-auth\n", domain, user);
-                ntlm_errno = NTLM_ERR_LOGON;
+                ntlm_errno = NtlmError::LoginEror;
                 return nullptr;
             }
         }
     }
 
     debug("checking domain: '%s', user: '%s', pass='%s'\n", domain, user, pass);
 
-    rv = SMB_Logon_Server(handle, user, pass, domain, 1);
+    const auto rv = SMB_Logon_Server(handle, user, pass, domain, 1);
     debug("Login attempt had result %d\n", rv);
 
-    if (rv != NTLM_ERR_NONE) {  /* failed */
-        ntlm_errno = rv;
+    switch (rv) {
+    case SMBlibE_Success:
+        ntlm_errno = NtlmError::None;
+        break;
+    case SMBlibE_BAD:
+        ntlm_errno = NtlmError::BlobError;
+        return nullptr;
+    case SMBlibE_ProtLow:
+    case SMBlibE_NoSpace:
+    case SMBlibE_BadParam:
+    case SMBlibE_NegNoProt:
+    case SMBlibE_LowerLayer:
+    case SMBlibE_SendFailed:
+    case SMBlibE_RecvFailed:
+    case SMBlibE_ProtUnknown:
+    case SMBlibE_NoSuchMsg:
+        ntlm_errno = NtlmError::ProtocolError;
+        return nullptr;
+    case SMBlibE_NotImpl:
+    case SMBlibE_CallFailed:
+    case SMBlibE_Remote:
+        ntlm_errno = NtlmError::ServerError;
+        return nullptr;
+    case SMBlibE_GuestOnly:
+        ntlm_errno = NtlmError::LoginEror;
+        return nullptr;
+    default:
+        ntlm_errno = NtlmError::ServerError;
```

Comment:
FWIW, this PR code hides/forgets the value returned by SMB_Logon_Server(). The PR description does not mention any functionality changes. I do not know whether this is a functionality change, but it may be worth double checking, especially if you are not testing this code (beyond building it).

Reply:
Double checked. It is not a functionality change; the caller uses SMB_Get_Last_Error to dig deeper into the reasons for the failure if it so chooses.